resources/services/s3/buckets.go

```diff
@@ -451,27 +453,53 @@ func S3Buckets() *schema.Table {
 // ====================================================================================================================
 //                                               Table Resolver Functions
 // ====================================================================================================================
+
 func fetchS3Buckets(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
 	svc := meta.(*client.Client).Services().S3
 	response, err := svc.ListBuckets(ctx, nil)
 	if err != nil {
 		return diag.WrapError(err)
 	}
-	wb := make([]*WrappedBucket, len(response.Buckets))
-	for i, b := range response.Buckets {
-		wb[i] = &WrappedBucket{b, nil, nil}
+
+	var wg sync.WaitGroup
+	buckets := make(chan types.Bucket)
+	errs := make(chan error)
+	for i := 0; i < fetchS3BucketsPoolSize; i++ {
+		wg.Add(1)
+		go fetchS3BucketsWorker(ctx, meta, buckets, errs, res, &wg)
 	}
+	for _, bucket := range response.Buckets {
+		buckets <- bucket
+	}
+	close(buckets)
+	go func() {
+		wg.Wait()
+		close(errs)
+	}()
 
-	res <- wb
-	return nil
+	var diags diag.Diagnostics
+	for err = range errs {
```

Comment:
`for err := range errs`, then do `diags.Add(err)` below?

Reply:
Thanks for catching this one, did too much changing around and missed that one.